virttest/gluster.py

```diff
@@ -319,25 +320,47 @@ def create_gluster_uri(params, stripped=False):
     """
     vol_name = params.get("gluster_volume_name")
 
+    # Access gluster server by unix domain socket
+    unix_socket = os.path.exists(params.get("gluster_server", ""))
+
     error_context.context("Host name lookup failed")
-    hostname = socket.gethostname()
     gluster_server = params.get("gluster_server")
-    gluster_port = params.get("gluster_port", "0")
-    if not gluster_server:
-        gluster_server = hostname
+    if not gluster_server or unix_socket:
+        gluster_server = socket.gethostname()
     if not gluster_server or gluster_server == "(none)":
-        if_up = utils_net.get_net_if(state="UP")
-        ip_addr = utils_net.get_net_if_addrs(if_up[0])["ipv4"][0]
-        gluster_server = ip_addr
+        gluster_server = utils_net.get_host_ip_address(params)
 
     # Start the gluster dameon, if not started
     # Building gluster uri
     gluster_uri = None
     if stripped:
         gluster_uri = "%s:/%s" % (gluster_server, vol_name)
     else:
-        gluster_uri = "gluster://%s:%s/%s/" % (gluster_server, gluster_port,
-                                               vol_name)
+        transport = ""
+        host = ""
+        volume = "/%s/" % vol_name
+
+        if unix_socket:
+            transport = "+unix"
+        else:
+            # Access gluster server by hostname or ip
+            gluster_transport = params.get("gluster_transport")
```

Comment:
Please add the document in `guest-hw.cfg` for all the newly introduced params.

Reply:
done